packages/http-crawler/src/internals/http-crawler.ts

```diff
@@ -0,0 +1,845 @@
+import { addTimeoutToPromise, tryCancel } from '@apify/timeout';
+import { concatStreamToBuffer, readStreamToString } from '@apify/utilities';
+import type {
+    AutoscaledPoolOptions,
+    BasicCrawlerOptions,
+    ErrorHandler,
+    RequestHandler,
+    CrawlingContext,
+    ProxyConfiguration,
+    Request,
+    Session,
+} from '@crawlee/basic';
+import {
+    BasicCrawler,
+    BASIC_CRAWLER_TIMEOUT_BUFFER_SECS,
+    CrawlerExtension,
+    mergeCookies,
+    Router,
+    validators,
+    Configuration,
+} from '@crawlee/basic';
+import type { Awaitable, Dictionary } from '@crawlee/types';
+import type { RequestLike, ResponseLike } from 'content-type';
+import contentTypeParser from 'content-type';
+import mime from 'mime-types';
+import type { OptionsInit, Method, Request as GotRequest, Response as GotResponse, GotOptionsInit } from 'got-scraping';
+import { gotScraping, TimeoutError } from 'got-scraping';
+import { extname } from 'path';
+import type { IncomingHttpHeaders, IncomingMessage } from 'http';
+import iconv from 'iconv-lite';
+import ow from 'ow';
+import util from 'util';
+
+/**
+ * Default mime types, which HttpScraper supports.
+ */
+const HTML_AND_XML_MIME_TYPES = ['text/html', 'text/xml', 'application/xhtml+xml', 'application/xml'];
+const APPLICATION_JSON_MIME_TYPE = 'application/json';
+const HTTP_OPTIMIZED_AUTOSCALED_POOL_OPTIONS: AutoscaledPoolOptions = {
+    desiredConcurrency: 10,
+    snapshotterOptions: {
+        eventLoopSnapshotIntervalSecs: 2,
+        maxBlockedMillis: 100,
+    },
+    systemStatusOptions: {
+        maxEventLoopOverloadedRatio: 0.7,
+    },
+};
+
+export type HttpErrorHandler<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = Dictionary,
+    > = ErrorHandler<HttpCrawlingContext<UserData, JSONData>>;
+
+export interface HttpCrawlerOptions<Context extends InternalHttpCrawlingContext = InternalHttpCrawlingContext> extends BasicCrawlerOptions<Context> {
+    /**
+     * An alias for {@apilink HttpCrawlerOptions.requestHandler}
```

Comment:
This is not an alias, it's deprecated I thought (and pending removal)

Reply:
https://github.com/apify/crawlee/blob/efe98280ee0e928e0eaa7512349c3ca37205a9bc/packages/http-crawler/src/internals/http-crawler.ts#L337-L344

It looks like an alias to me :P You're right I forgot to put `@deprecated` on it

Reply:
its there only for back compatibility/easier upgrading/less friction

i guess we need to keep this because it handles the compat for cheerio crawler, right? otherwise, we dont need to care about this for new crawlers.

Reply:
Yes, it's there for compat.